src/components/Avatar/Avatar.scss

```diff
@@ -0,0 +1,157 @@
+@use '../variables';
+
+$block: '.#{variables.$ns-new}avatar';
+
+#{$block} {
+    overflow: hidden;
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+    border-radius: 50%;
+    background-color: var(--g-color-base-background);
+
+    &__image {
+        display: block;
+        width: 100%;
+        height: 100%;
+        object-fit: cover;
+    }
+
+    &__icon > svg {
+        display: block;
+    }
+
+    &__text {
+        font-weight: 500;
+    }
+
+    &_withBorder,
+    &_view_outlined {
+        position: relative;
+
+        &::before,
+        &::after {
+            content: '';
+            z-index: 1;
+            position: absolute;
+            inset: 0;
+            border-radius: 50%;
+        }
+
+        &::before {
+            border: 3px solid var(--g-color-base-background);
+        }
+
+        &::after {
+            border: 2px solid currentColor;
+        }
+    }
+
+    &_size {
+        &_3xs {
+            width: 17px;
+            height: 17px;
+        }
+
+        &_2xs {
+            width: 20px;
+            height: 20px;
+        }
+
+        &_xs {
+            width: 24px;
+            height: 24px;
+        }
+
+        &_s {
+            width: 28px;
+            height: 28px;
+        }
+
+        &_m {
+            width: 32px;
+            height: 32px;
+        }
+
+        &_l {
+            width: 42px;
+            height: 42px;
+        }
+
+        &_xl {
+            width: 50px;
+            height: 50px;
+        }
+
+        &_3xs,
+        &_2xs,
+        &_xs,
+        &_s {
+            #{$block}__text {
+                font-size: var(--g-text-caption-1-font-size);
+                line-height: var(--g-text-caption-1-line-height);
+            }
+        }
+
+        &_m,
+        &_l {
+            #{$block}__text {
+                font-size: var(--g-text-body-1-font-size);
+                line-height: var(--g-text-body-1-line-height);
+            }
+        }
+
+        &_xl {
+            #{$block}__text {
+                font-size: var(--g-text-body-2-font-size);
+                line-height: var(--g-text-body-2-line-height);
+            }
+        }
+    }
+
+    &_theme {
+        &_normal {
+            &#{$block}_view_filled {
+                background-color: var(--g-color-base-misc-light);
+                color: unset;
+
+                #{$block}__icon,
+                #{$block}__text {
+                    color: var(--g-color-text-misc);
+                }
+            }
+
+            &#{$block}_view_outlined {
+                background-color: var(--g-color-base-background);
+                color: var(--g-color-text-misc);
+
+                #{$block}__icon,
+                #{$block}__text {
+                    color: var(--g-color-text-misc);
```

Comment:
This could be done with private css variables, see [example](https://github.com/gravity-ui/uikit/blob/next/src/components/Button/Button.scss#L241-L245)